src/plugins/intel_cpu/src/nodes/multinomial.hpp

```diff
@@ -0,0 +1,178 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <random>
+#include <string>
+
+#include "ie_common.h"
+#include "ie_parallel.hpp"
+#include "node.h"
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+class Multinomial : public Node {
+public:
+    Multinomial(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context);
+
+    void getSupportedDescriptors() override;
+    void initSupportedPrimitiveDescriptors() override;
+    std::string getPrimitiveDescriptorType() const override;
+
+    bool created() const override;
+
+    static bool isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept;
+
+    bool needPrepareParams() const override;
+    void prepareParams() override;
+
+    bool isExecutable() const override;
+    void execute(dnnl::stream strm) override;
+    void executeDynamicImpl(dnnl::stream strm) override;
+    bool canBeInPlace() const override {
+        return false;
+    }
+
+protected:
+    bool needShapeInfer() const override;
+
+private:
+    /// Multinomial params
+    bool m_with_replacement = false;
+    bool m_log_probs = false;
+    uint64_t m_global_seed = 0;
```

Comment:
Is global seed used? From what I see, mt19937 just uses m_op_seed

Reply:
Not used, kept to match spec & for the near future when RandomUniform is introduced here instead of mersenne

Reply:
If not used it just consume memory. It would be add when is required and add this information for `RandomUniform` task to added.

Reply:
Replaced with updated seed generation to match expected behavior, now both are used